src/machine.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+use std::mem::align_of;
+
+use rustc::mir::{Mir, Local};
+
+use interp::TypedVal;
+
+const X64_ALIGN_SIZE: u8 = 8;
+const X32_ALIGN_SIZE: u8 = 4;
+
+#[derive(Debug)]
+struct Frame<'tcx> {
+    // MIR is a CFG of a single function in rust, so it makes sense
+    // for this to be in the frame.
+    mir: Mir<'tcx>,
+
+    // FIXME: Hacky... I can't find a way to actually access the private u32
+    // inside the `Local` tuple struct. `Local` does  implement the Hashable
+    // and PartialEQ trait however, so for  now it might just work to to use it
+    // as the key
+    locals: HashMap<Local, Vec<u8>>
+}
+
+impl<'tcx> Frame<'tcx> {
+    fn new(mir: Mir<'tcx>) -> Frame {
+        Frame {
+            locals: HashMap::new(),
```

Comment:
Not that you need to do it in this PR, but you might want to lazily allocate this list and/or not allocate it if there are no local variables. Some small closures in tight loops won't have any locals, and that sort of saving can become useful.

Reply:
That's a good idea, I will probably do this in a future PR. I've added a TODO comment so that I don't forget this

Reply:
Again, if we raise an issue for all of the things that we want to implement in later PRs, that would be great.

Reply:
That sounds sensible.